Feature/vvereen direct lake docs (#3057)
MicrosoftDocs/fabric-docs: docs/fundamentals/direct-lake-how-it-works.md

## DirectQuery fallback

When using Direct Lake on SQL endpoints, a query sent to a Direct Lake semantic model can fall back to [DirectQuery mode](/power-bi/connect-data/service-dataset-modes-understand) in which case the table no longer operates in Direct Lake mode. It retrieves data directly from the SQL analytics endpoint of the lakehouse or warehouse. Such queries always return the latest data because they're not constrained to the point in time of the last framing operation. However, fallback operations might result in slower query performance.

> [!IMPORTANT]
> If possible, you should always design your solution—or size your capacity—to avoid DirectQuery fallback. That's because it might result in slower query performance.

> Direct Lake on OneLake does not support DirectQuery fallback and operates only in DirectLakeOnly mode. If you need to avoid fallback entirely, consider using Direct Lake on OneLake. Direct Lake on Onelake is the recommended Direct Lake flavor for new semantic model applications.

Reviewer: >If you need to avoid fallback entirely, consider using Direct Lake on OneLake.

a bit weird sentence if i am being honest, best if we remove it entirely, in this case less is more :)

Author: Agreed, we can remove it.

### When Direct Lake mode is used

Queries use Direct Lake mode only when *all* of the following conditions are satisfied:

> - No tables referenced by the semantic model have SQL [row-level security (RLS)](direct-lake-security-integration.md) defined at the SQL analytics endpoint.

Reviewer: those are conditions for using VertiPlans in DirectLake SQL.. it miscommunicates things to customers.

Author: Would naming the section "Conditions for avoiding DirectQuery fallback" be better?

- No tables referenced by the semantic model have SQL dynamic data masking (DDM) defined at the SQL analytics endpoint.
- No tables referenced by the semantic model have SQL [object-level security (OLS)](direct-lake-security-integration.md) defined at the SQL analytics endpoint.
- The semantic model does not reference any tables based on unmaterialized SQL views.
- No single table in the semantic model exceeds guardrail limits as defined in [Fabric capacity requirements](direct-lake-overview.md#fabric-capacity-requirements):
  - Number of Parquet files is within limit
  - Number of row groups is within limit
  - Number of rows is within limit
- The semantic model has been refreshed (framed) since the underlying Delta tables were created or modified.

> - The capacity is not under memory pressure.

Reviewer: not true

Author: I can remove rows 100 & 101, are the rest are still accurate? Apologies pulled these two from community blog posts.

## DirectQuery fallback

When using Direct Lake on SQL endpoints, a query sent to a Direct Lake semantic model can fall back to [DirectQuery mode](/power-bi/connect-data/service-dataset-modes-understand) in which case the table no longer operates in Direct Lake mode. It retrieves data directly from the SQL analytics endpoint of the lakehouse or warehouse. Such queries always return the latest data because they're not constrained to the point in time of the last framing operation. However, fallback operations might result in slower query performance.

> [!IMPORTANT]
> If possible, you should always design your solution—or size your capacity—to avoid DirectQuery fallback. That's because it might result in slower query performance.

Direct Lake on OneLake does not support DirectQuery fallback and operates only in DirectLakeOnly mode. If you need to avoid fallback entirely, consider using Direct Lake on OneLake. Direct Lake on Onelake is the recommended Direct Lake flavor for new semantic model applications.

### When Direct Lake mode is used

> In Automatic mode, queries use Direct Lake mode only when *all* of the following conditions are satisfied:

Reviewer 2: ```suggestion
Queries use Direct Lake mode only when *all* of the following conditions are satisfied:
```

- No tables referenced by the semantic model have SQL [row-level security (RLS)](direct-lake-security-integration.md) defined at the SQL analytics endpoint.
- No tables referenced by the semantic model have SQL dynamic data masking (DDM) defined at the SQL analytics endpoint.
- No tables referenced by the semantic model have SQL [object-level security (OLS)](direct-lake-security-integration.md) defined at the SQL analytics endpoint.
- The semantic model does not reference any tables based on unmaterialized SQL views.
- No single table in the semantic model exceeds guardrail limits:
    - Number of Parquet files is within limit
    - Number of row groups is within limit
    - Number of rows is within limit

>   For limits by SKU, see [Fabric capacity requirements](direct-lake-overview.md#fabric-capacity-requirements).

Reviewer 2: ```suggestion
- No single table in the semantic model exceeds guardrail limits as defined in [Fabric capacity requirements](direct-lake-overview.md#fabric-capacity-requirements):
  - Number of Parquet files is within limit
  - Number of row groups is within limit
  - Number of rows is within limit
```

- The capacity is not under memory pressure.

A single table exceeding any guardrail can prevent Direct Lake mode for the entire model.

### What happens when conditions are not met

The behavior depends on the **DirectLakeBehavior** setting:

- **Automatic** (default): The query silently falls back to DirectQuery mode. Reports continue to work, but performance might be slower.
- **DirectLakeOnly**: The query fails with an error. 

Use DirectLakeOnly mode during development to surface these errors early and fix the root cause before users encounter silent performance degradation.

### Control fallback with DirectLakeBehavior

You can control fallback of your Direct Lake semantic models by setting the **DirectLakeBehavior** property. This setting only applies to Direct Lake on SQL endpoints.

| Value | Description |
| --- | --- |
| **Automatic** | (Default) Falls back to DirectQuery mode silently when conditions are not met. |
| **DirectLakeOnly** | Errors out instead of falling back. Use during development. |
| **DirectQueryOnly** | Always uses DirectQuery mode. Use to measure fallback performance. |

> 

Reviewer 2: ```suggestion
### Control fallback with DirectLakeBehavior

When Direct Lake conditions aren't met, the behavior of your semantic models depends on the **DirectLakeBehavior** setting. This setting only applies to Direct Lake on SQL endpoints.

You can set the **DirectLakeBehavior** property to the following three values:

| Value | Description |
| --- | --- |
| **Automatic** | (Default) If conditions aren't met, the query silently falls back to DirectQuery mode. Reports continue to work, but performance might be slower. Use this mode in production to preserve functionality for your users. |
| **DirectLakeOnly** | If conditions aren't met, the query fails with an error. Use this mode during development to identify and resolve errors. |
| **DirectQueryOnly** | The query always uses DirectQuery mode. Use to this mode during development to measure fallback performance. |

```
